packages/components/src/components/multi-select/multi-select/multi-select.tsx

```diff
@@ -0,0 +1,334 @@
+
+  /** Disables the multi-select */
+  @Prop() public disabled?: boolean = false;
+
+  /** A Boolean attribute indicating that an option with a non-empty string value must be selected. */
+  @Prop() public required?: boolean = false;
+
+  /** Changes the direction to which the dropdown list appears. */
+  @Prop() public dropdownDirection?: MultiSelectDropdownDirection = 'auto';
+
+  /** Adapts the select color depending on the theme. */
+  @Prop() public theme?: Theme = 'light';
+
+  @State() private selectedString = '';
+  @State() private isOpen = false;
+
+  private nativeSelect: HTMLSelectElement = document.createElement('select');
+  private multiSelectOptions: HTMLPMultiSelectOptionElement[] = [];
+  private inputContainer: HTMLDivElement;
+  private inputElement: HTMLInputElement;
+  private listElement: HTMLUListElement;
+  private isWithinForm: boolean;
+
+  @Listen('internalOptionUpdate')
+  public updateOptionHandler(): void {
+    if (this.isWithinForm) {
+      updateNativeSelectOptions(this.nativeSelect, this.multiSelectOptions);
+    }
+    this.updateSelectedString();
+  }
+
+  public connectedCallback(): void {
+    this.isWithinForm = isWithinForm(this.host);
+    if (this.isWithinForm) {
+      syncNativeSelect(this.nativeSelect, this.host, this.name, this.disabled, this.required);
+    }
+  }
+
+  public componentWillLoad(): void {
+    document.addEventListener('mousedown', this.onClickOutside, true);
+  }
+
+  public componentDidLoad(): void {
+    this.updateOptions();
+  }
+
+  public componentShouldUpdate(newVal: unknown, oldVal: unknown): boolean {
+    return hasPropValueChanged(newVal, oldVal);
+  }
+
+  public disconnectedCallback(): void {
+    document.removeEventListener('mousedown', this.onClickOutside, true);
+  }
+
+  public render(): JSX.Element {
+    validateProps(this, propTypes);
+    attachComponentCss(
+      this.host,
+      getComponentCss,
+      this.nativeSelect.selectedOptions.length > 0,
+      getDropdownDirection(this.dropdownDirection, this.inputContainer, this.multiSelectOptions),
+      this.isOpen,
+      this.disabled,
+      this.hideLabel,
+      this.state,
+      this.isWithinForm,
+      this.theme
+    );
+    syncMultiSelectOptionProps(this.multiSelectOptions, this.theme);
+
+    const PrefixedTagNames = getPrefixedTagNames(this.host);
+    const dropdownId = 'list';
+
+    return (
+      <Host>
+        <div class="root">
+          <label class="label" id="label">
+            {this.selectedString && (
+              <span class="sr-text">{getSelectedOptions(this.multiSelectOptions).length} options selected</span>
+            )}
+            {!this.hideLabel && hasLabel(this.host, this.label) && (
+              <span class="label__text" onClick={() => this.inputElement.focus()}>
+                {this.label || <slot name="label" />}
+                {isRequiredAndParentNotRequired(this.host, this.nativeSelect) && <Required />}
```

Comment:
Reading from synched `this.nativeSelect` doesn't seem right especially since it might be undefined?